examples/cpp/main.cpp

```diff
@@ -0,0 +1,30 @@
+#include <stdio.h>
+
+#include <wizer.h>
+
+class Test {
+    public:
+        Test() : value(1) {}
+        ~Test() {
+            printf("global destructor\n");
+        }
+
+        int value;
+};
+
+int orig_value = 0;
+Test t;
+
+static void init_func() {
+    // This should run after the ctor for `t`, and before `main`.
+    orig_value = t.value;
+    t.value = 2;
+}
+
+WIZER_INIT(init_func);
+
+int main(int argc, char** argv) {
```

Comment:
It might be good to have the example demonstrate how to run the same module without wizerization. This works as expected for me:

```suggestion
        Test() : value(1) {
            printf("global constructor (should be the first printed line)\n");
        }
        ~Test() {
            printf("global destructor (should be the last printed line)\n");
        }

        int value;
};

bool initialized = false;
int orig_value = 0;
Test t;

static void init_func() {
    // This should run after the ctor for `t`, and before `main`.
    orig_value = t.value;
    t.value = 2;
    initialized = true;
}

WIZER_INIT(init_func);

int main(int argc, char** argv) {
    if (!initialized) init_func();
```

Reply:
Good idea; applied!